packages/quantic/force-app/custom-token/main/default/classes/CustomTokenProvider.cls

```diff
@@ -0,0 +1,108 @@
+        return userIdentities;
+    }
+
+    private static Boolean isTokenSettingInvalid(TokenSettings__c settings) {
+        return  String.isEmpty(settings.OrganizationId__c) ||
+                !CustomTokenProvider.isValidURL(settings.PlatformUrl__c) ||
+                String.isEmpty(settings.ApiKey__c);
+    }
+
+    private static Boolean isASuccessResponse(HTTPResponse response) {
+        Integer responseStatusCode = response.getStatusCode();
+        return (responseStatusCode >= 200 && responseStatusCode < 300);
+    }
+
+    private static String extractTokenFromResponse(HTTPResponse response) {
+        Map<String,Object> responseAsMap = (Map<String, Object>) JSON.deserializeUntyped(response.getBody());
+        return (String) responseAsMap.get('token');
+    }
+
+    /**
+     * Build a HttpRequest to get a search token from a TokenSettings__c object.
+     * @param {settings} the settings to use.
+     */
+    private static HttpRequest buildHttpRequest(TokenSettings__c settings) {
+        HttpRequest tokenRequest = new HttpRequest();
+        tokenRequest.setEndpoint(settings.PlatformUrl__c + TOKEN_PATH + '?organizationId=' + settings.OrganizationId__c);
+        tokenRequest.setMethod('POST');
+        tokenRequest.setHeader('Accept', CONTENT_TYPE_APPLICATION_JSON);
+        tokenRequest.setHeader('Content-Type', CONTENT_TYPE_APPLICATION_JSON);
+        tokenRequest.setHeader('Authorization', 'Bearer ' + settings.ApiKey__c);
+
+        Map<String,Object> requestBody = new Map<String, Object>();
+        requestBody.put('userIds', CustomTokenProvider.getUserIdentities());
+        String requestBodySerialized = JSON.serialize(requestBody);
+        tokenRequest.setBody(requestBodySerialized);
+        tokenRequest.setHeader('Content-Length', String.valueOf(requestBodySerialized.length()));
+
+        return tokenRequest;
+    }
+
+    /**
+     * Get a search token from Coveo Platform based on the setting and the current user identity.
+     * @param {settings} the settings to use.
+     */
+    private static String getToken(TokenSettings__c settings) {
+        HttpRequest tokenRequest = CustomTokenProvider.buildHttpRequest(settings);
+
+        HTTPResponse tokenResponse = (new Http()).send(tokenRequest);
+        if(!CustomTokenProvider.isASuccessResponse(tokenResponse)) {
+            throw new HttpException(tokenResponse.getBody());
+        }
+
+        return CustomTokenProvider.extractTokenFromResponse(tokenResponse);
+    }
+
+    /**
+     * Get a serialized object representing a HeadlessConfiguration, based on the TokenSettings__c
+     * of the organization and the user identity of the user that called the method/loaded the page.
+     */
+    @AuraEnabled(continuation=true cacheable=false)
+    global static String getHeadlessConfiguration() {
+        TokenSettings__c settings = TokenSettings__c.getOrgDefaults();
+
+        if (CustomTokenProvider.isTokenSettingInvalid(settings)) {
```

Comment:
I'd remove this part. Lets say someone mess up the values in prod for a customer, I'd prefer that the components stop working and display and error instead of showing "demo data".

Reply:
Okok. I'll throw an Exception.
It'll be left to the person who will plug those classes in the frontend to handle it tho.